server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -515,4 +520,89 @@ public void testDropPrimaryDuringReplication() throws Exception {
             verifyStoreContent();
         }
     }
+
+    public void testWaitUntilRefresh() throws Exception {
+        final String primaryNode = internalCluster().startNode();
+        createIndex(INDEX_NAME);
+        ensureYellowAndNoInitializingShards(INDEX_NAME);
+        final String replicaNode = internalCluster().startNode();
+        ensureGreen(INDEX_NAME);
+        final int initialDocCount = scaledRandomIntBetween(600, 700);
```

Comment:
nit: Any specific reason for selecting docs b/w 600 to 700 ? Is it better to ingest more documents around 2k to 5k, I have seen issues happen more frequently where there are more docs

Reply:
Sure I can ingest more docs. I will update that